ceph/rados/core_workflows.py

```diff
@@ -3151,3 +3151,69 @@ def get_osd_perf_dump(self, osd_id):
         """
         base_cmd = f"ceph tell osd.{osd_id} perf dump"
         return self.run_ceph_command(cmd=base_cmd)
+
+    def get_available_ssd(self, node):
+        """
+        Method returns the available SSD list in the provided node.
+            Args:
+                 rados_object: RadosOrchestrator class object
+                 node: node name
+
+             Returns: Returns the available device path list of the node.
+
+        """
+        ssd_paths = []
+        available_device_list = self.get_orch_device_list(node)
+        for path_list in available_device_list[0]["devices"]:
+            if path_list["human_readable_type"] == "hdd":
+                continue
+            if (
+                path_list["human_readable_type"] == "ssd"
+                and path_list["available"] is True
+            ):
+                ssd_paths.append(path_list["path"])
+        return ssd_paths
+
+    def get_available_hdd(self, node):
+        """
+        Method returns the available HDD list in the provided node.
+            Args:
+                 rados_object: RadosOrchestrator class object
+                 node: node name
+
+             Returns: Returns the available device path list of the node.
+
+        """
+        hdd_paths = []
+        available_device_list = self.get_orch_device_list(node)
+        for path_list in available_device_list[0]["devices"]:
+            if path_list["human_readable_type"] == "sdd":
+                continue
+            if (
+                path_list["human_readable_type"] == "hdd"
+                and path_list["available"] is True
+            ):
+                hdd_paths.append(path_list["path"])
+        return hdd_paths
+
+    def is_path_available(self, node, path):
+        """
+        Method returns the True or False based of device path availability.
+            Args:
+                rados_object: RadosOrchestrator class object
+                node: node name
+                path : device path
+                       Example: /dev/sdb
+            Returns: Return the True or False
+        """
+        available_device_list = self.is_path_available(node)
```

Comment:
Is the method supposed to call itself here?
I believe it's a typo and should be changed to `self.get_orch_device_list(node)`

Reply:
Modified the code.